asset-transfer-sbe/chaincode-java/src/main/java/org/hyperledger/fabric/samples/sbe/AssetContract.java

```diff
@@ -0,0 +1,196 @@
+import org.hyperledger.fabric.shim.ChaincodeStub;
+import org.hyperledger.fabric.shim.ext.sbe.StateBasedEndorsement;
+import org.hyperledger.fabric.shim.ext.sbe.impl.StateBasedEndorsementFactory;
+
+import com.owlike.genson.Genson;
+
+@Contract(
+        name = "sbe",
+        info = @Info(
+                title = "Asset Contract",
+                description = "Asset Transfer Smart Contract, using State Based Endorsement(SBE), implemented in Java",
+                version = "0.0.1-SNAPSHOT",
+                license = @License(
+                        name = "Apache 2.0 License",
+                        url = "http://www.apache.org/licenses/LICENSE-2.0.html")))
+@Default
+public final class AssetContract implements ContractInterface {
+
+    private final Genson genson = new Genson();
+
+    private enum AssetTransferErrors {
+        ASSET_NOT_FOUND,
+        ASSET_ALREADY_EXISTS
+    }
+
+    /**
+     * Creates a new asset.
+     * Sets the endorsement policy of the assetId Key, such that current owner Org Peer is required to endorse future updates.
+     *
+     * @param ctx the transaction context
+     * @param assetId the ID of the new asset
+     * @param value the value of the new asset
+     * @param owner the owner of the new asset
+     * @return the created asset
+     */
+    @Transaction(intent = Transaction.TYPE.SUBMIT)
+    public Asset CreateAsset(final Context ctx, final String assetId, final int value, final String owner) {
+        ChaincodeStub stub = ctx.getStub();
+
+        if (AssetExists(ctx, assetId)) {
+            String errorMessage = String.format("Asset %s already exists", assetId);
+            System.out.println(errorMessage);
+            throw new ChaincodeException(errorMessage, AssetTransferErrors.ASSET_ALREADY_EXISTS.toString());
+        }
+
+        final String ownerOrg = getClientOrgId(ctx);
+        Asset asset = new Asset(assetId, value, owner, ownerOrg);
+        String assetJSON = genson.serialize(asset);
+        stub.putStringState(assetId, assetJSON);
+
+        setAssetStateBasedEndorsement(ctx, assetId, new String[]{ownerOrg});
+
+        return asset;
+    }
+
+    /**
+     * Retrieves an asset with the given assetId.
+     *
+     * @param ctx the transaction context
+     * @param assetId the ID of the asset
+     * @return the asset found on the ledger if there was one
+     */
+    @Transaction(intent = Transaction.TYPE.EVALUATE)
+    public Asset ReadAsset(final Context ctx, final String assetId) {
+        ChaincodeStub stub = ctx.getStub();
+        String assetJSON = stub.getStringState(assetId);
+
+        if (assetJSON == null || assetJSON.isEmpty()) {
+            String errorMessage = String.format("Asset %s does not exist", assetId);
+            System.out.println(errorMessage);
+            throw new ChaincodeException(errorMessage, AssetTransferErrors.ASSET_NOT_FOUND.toString());
+        }
+
+        Asset asset = genson.deserialize(assetJSON, Asset.class);
```

Comment:
Hmmm, when I read the asset I get:
{"owner":"Org1User1","ownerOrg":"Org1MSP","value":100}

No assetID.